### PR TITLE
Fix calls of non constexpr functions inside static_assert in std::complex tests

### PR DESCRIPTION
**Pull request by SergeyKopienko**
The root cause is Microsoft implementation of C++ standard library declares some ``std::complex`` related functions just ``inline`` prior C++20 while they should be ``constexpr``

### CODE REVIEW

File: test/xpu_api/numerics/complex.number/cmplx.over/proj.pass.cpp [hunk 1]
```diff
@@ -22,15 +22,15 @@ template <class T>
 void
 test(T x, typename std::enable_if<std::is_integral<T>::value>::type* = 0)
 {
-    static_assert((std::is_same<decltype(dpl::proj(x)), dpl::complex<double> >::value), "");
+    STD_COMPLEX_TESTS_STATIC_ASSERT((std::is_same<decltype(dpl::proj(x)), dpl::complex<double> >::value), "");
```
Comment (rarutyun) [inline test/xpu_api/numerics/complex.number/cmplx.over/proj.pass.cpp:25]: All tests with `decltype` should work independently of `constexpr`. So, all those lines can be used in `static_assert` expression.

File: test/support/test_complex.h [hunk 2]
```diff
@@ -26,6 +26,24 @@
 #include <type_traits>
 #include <cassert>
 
+#if !_MSVC_STL_CPP17_COMPLEX_CONSTEXPR_FNC_BROKEN
+#    define STD_COMPLEX_TESTS_STATIC_ASSERT(...) static_assert(__VA_ARGS__)
+#else
+template <typename T>
+void
+__complex_tests_assert(T condition)
```
Comment (rarutyun) [inline test/support/test_complex.h:34]: Seems like you don't have the `static_assert` without message (I count empty message as a "message"). Probably you don't need this function at all.

File: test/support/test_complex.h [hunk 3]
```diff
@@ -26,6 +26,24 @@
...
+{
+    assert(condition);
+}
+template <typename T>
+void
+__complex_tests_assert(T condition, const char*)
```
Comment (rarutyun) [inline test/support/test_complex.h:40]: I would suggest to rewrite the function to function-like macro in this case to be able to see more informative message from `assert` if something is wrong.

File: test/support/test_config.h [hunk 4]
```diff
@@ -106,4 +106,10 @@
 
 #define TEST_HAS_NO_INT128
 
+#if defined(_MSC_VER) && __cplusplus <= 201703L
+#    define _MSVC_STL_CPP17_COMPLEX_CONSTEXPR_FNC_BROKEN 1
```
Comment (rarutyun) [inline test/support/test_config.h:110]: Please consider:
`define _MSVC_STL_CPP17_COMPLEX_CONSTEXPR_FNC_BROKEN (defined(_MSC_VER) && __cplusplus <= 201703L)`

Please check if it works

Comment (SergeyKopienko) [reply test/support/test_config.h]: I checked and use this variant:
#define _PSTL_MSVC_CPP17_COMPLEX_CONSTEXPR_FNC_BROKEN (_MSC_VER && __cplusplus <= 201703L)

File: test/support/test_complex.h (hunk 3 above)
Comment (SergeyKopienko) [reply test/support/test_complex.h]: Fixed.

**Comment by rarutyun** [comment]
I've made small naming improvement for the broken macro. Everything else seems ok. Waiting for CI to merge it.

**Review by rarutyun — APPROVED** [review]

**Pull request merged**